numpy/lib/npyio.pyi

```diff
@@ -1,98 +1,249 @@
+# returns an `ndarray`/`memmap` otherwise
+def load(
+    file: str | bytes | os.PathLike[Any] | IO[bytes],
+    mmap_mode: L[None, "r+", "r", "w+", "c"] = ...,
+    allow_pickle: bool = ...,
+    fix_imports: bool = ...,
+    encoding: L["ASCII", "latin1", "bytes"] = ...,
+) -> Any: ...
+
+def save(
+    file: str | os.PathLike[str] | IO[bytes],
+    arr: ArrayLike,
+    allow_pickle: bool = ...,
+    fix_imports: bool = ...,
+) -> None: ...
+
+def savez(
+    file: str | os.PathLike[str] | IO[bytes],
+    *args: ArrayLike,
+    **kwds: ArrayLike,
+) -> None: ...
+
+def savez_compressed(
+    file: str | os.PathLike[str] | IO[bytes],
+    *args: ArrayLike,
+    **kwds: ArrayLike,
+) -> None: ...
+
+@overload
+def loadtxt(
+    fname: str | os.PathLike[str] | IO[Any],
+    dtype: None = ...,
+    comments: str | Sequence[str] = ...,
+    delimiter: None | str = ...,
+    converters: None | Mapping[int | str, Callable[[str], Any]] = ...,
+    skiprows: int = ...,
+    usecols: int | Sequence[int] = ...,
+    unpack: bool = ...,
+    ndmin: L[0, 1, 2] = ...,
+    encoding: None | str = ...,
+    max_rows: None | int = ...,
+    *,
+    like: None | ArrayLike = ...
+) -> NDArray[float64]: ...
+@overload
+def loadtxt(
+    fname: str | os.PathLike[str] | IO[Any],
+    dtype: _DTypeLike[_SCT],
+    comments: str | Sequence[str] = ...,
+    delimiter: None | str = ...,
+    converters: None | Mapping[int | str, Callable[[str], Any]] = ...,
+    skiprows: int = ...,
+    usecols: int | Sequence[int] = ...,
+    unpack: bool = ...,
+    ndmin: L[0, 1, 2] = ...,
+    encoding: None | str = ...,
+    max_rows: None | int = ...,
+    *,
+    like: None | ArrayLike = ...
+) -> NDArray[_SCT]: ...
+@overload
 def loadtxt(
-    fname,
-    dtype=...,
-    comments=...,
-    delimiter=...,
-    converters=...,
-    skiprows=...,
-    usecols=...,
-    unpack=...,
-    ndmin=...,
-    encoding=...,
-    max_rows=...,
+    fname: str | os.PathLike[str] | IO[Any],
+    dtype: DTypeLike,
+    comments: str | Sequence[str] = ...,
+    delimiter: None | str = ...,
+    converters: None | Mapping[int | str, Callable[[str], Any]] = ...,
+    skiprows: int = ...,
+    usecols: int | Sequence[int] = ...,
+    unpack: bool = ...,
+    ndmin: L[0, 1, 2] = ...,
+    encoding: None | str = ...,
+    max_rows: None | int = ...,
     *,
-    like=...,
-): ...
+    like: None | ArrayLike = ...
+) -> NDArray[Any]: ...
+
 def savetxt(
-    fname,
-    X,
-    fmt=...,
-    delimiter=...,
-    newline=...,
-    header=...,
-    footer=...,
-    comments=...,
-    encoding=...,
-): ...
-def fromregex(file, regexp, dtype, encoding=...): ...
+    fname: str | os.PathLike[str] | IO[Any],
+    X: ArrayLike,
+    fmt: str | Sequence[str] = ...,
+    delimiter: str = ...,
+    newline: str = ...,
+    header: str = ...,
+    footer: str = ...,
+    comments: str = ...,
+    encoding: None | str = ...,
+) -> None: ...
+
+@overload
+def fromregex(
+    file: str | IO[Any],
```

Comment:
`fromregex` is currently the only file-manipulation-function that does not support path-like objects.
Might be something to add in a future PR?

Reply:
Maybe open an issue so it doesn't get lost.

Reply:
Good call, I just created one at https://github.com/numpy/numpy/issues/19679.